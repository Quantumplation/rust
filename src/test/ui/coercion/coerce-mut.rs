fn f(x: &mut i32) {}

fn main() {
    let x = 0;
    f(&x);
    //~^ ERROR arguments to this function are incorrect
    //~| expected mutable reference `&mut i32`
    //~| found reference `&{integer}`
    //~| types differ in mutability
}
