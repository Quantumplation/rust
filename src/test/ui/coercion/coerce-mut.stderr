error[E0308]: arguments to this function are incorrect
  --> $DIR/coerce-mut.rs:5:5
   |
LL |     f(&x);
   |     ^^--^
   |       |
   |       expected `&mut i32`, found `&{integer}`
   |
note: function defined here
  --> $DIR/coerce-mut.rs:1:4
   |
LL | fn f(x: &mut i32) {}
   |    ^ -----------
help: provide an argument of the correct type
   |
LL |     f({&mut i32});
   |     ^^^^^^^^^^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.
